#VRML_SIM R2021b utf8
WorldInfo {
  info [
    "This demonstration shows a DJI Mavic 2 PRO flying over a rural area."
    "After the drone has taken off, you can control it with the keyboard of your computer."
  ]
  title "DJI Mavic 2 PRO demo"
  basicTimeStep 8
  defaultDamping Damping {
    linear 0.5
    angular 0.5
  }
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.1571322551049566 -0.9775164133124283 -0.1406098009046198 4.795231080030554
  position 9.512301790284218 3.703785268314692 1.0706332129406755
  near 0.2
  follow "Mavic 2 PRO"
  followSmoothness 0.2
}
TexturedBackground {
  luminosity 3
}
TexturedBackgroundLight {
}
Floor {
  size 400 400
  tileSize 6 6
  appearance SandyGround {
  }
}
AdvertisingBoard {
  translation -7.11 0.189999 0.92
  rotation 1 0 0 -1.5707953071795862
  frontTexture [
    "C:/Users/halil/my_files/drone_simulation/qr1.jpg"
  ]
  displayWidth 1.5
  displayHeight 1.5
}
Panel {
  translation -18.9584 0 -7.19276
  rotation 0 -1 0 -1.3090053071795866
  size 5 7 0.1
}
Road {
  translation -43.8 0.01 -9.2
  rotation 0 1 0 1.1781
  width 3
  numberOfLanes 1
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    0 0 100
    200 0 200
  ]
  splineSubdivision 8
  appearance Pavement {
    type "black stone"
  }
}
Windmill {
  translation -43.8656 0 19.8433
  rotation 0 1 0 -1.0472
}
Windmill {
  translation -44.258 0 27.3356
  rotation 0 1 0 -1.0472
  name "windmill(1)"
}
Windmill {
  translation -46.2448 0 -30.573
  rotation 0 1 0 -1.0472
  name "windmill(2)"
}
Windmill {
  translation -38.75 0 -23.4
  rotation 0 1 0 -1.5708
  name "windmill(3)"
}
SmallManor {
  translation -50.35 0 -11.25
  rotation 0 1 0 4.32
}
SquareManhole {
  translation 0 -0.03 0
}
CardboardBox {
  translation -0.730157 0.3 1.22891
  rotation 0 1 0 -0.261795307179586
}
TeslaModel3Simple {
  translation -41.5139 0.31 -4.34169
  rotation 0 -1 0 -1.309
}
Pine {
  translation -14.0083 0 14.4828
  rotation 0 1 0 3.14159
}
Pine {
  translation -19.6195 0 24.3783
  rotation 0 1 0 2.87979
  name "pine tree(1)"
}
Pine {
  translation -10.6926 0 25.1339
  name "pine tree(4)"
}
Pine {
  translation -14.3441 0 -14.5621
  name "pine tree(5)"
}
Pine {
  translation -26.6328 0 7.1707
  name "pine tree(6)"
}
Pine {
  translation -9.37029 0 -14.0209
  rotation 0 1 0 -2.618
  name "pine tree(2)"
}
Forest {
  translation 0 -0.01 0
  shape [
    200 200
    200 0
    50 0
    50 50
    -50 50
    -50 -50
    50 -50
    50 0
    200 0
    200 -200
    -200 -200
    -200 200
  ]
  density 0.01
  type "cherry tree"
  randomSeed 1
  maxHeight 12
  minHeight 6
  maxRadius 6
  minRadius 3
}
Mavic2Pro {
  translation -5.62653e-08 0.07 -5.26855e-14
  rotation 0 -0.7071067811865475 -0.7071067811865475 -3.14159
  bodySlot [
    Accelerometer {
    }
    DistanceSensor {
      rotation -0.5773502691890485 -0.5773492691885823 0.5773512691895147 -2.094395307179586
      name "ds_down"
      lookupTable [
        0 0 0
        2 1000 0
      ]
      type "sonar"
      aperture 1.5709
    }
    DistanceSensor {
      translation 0 0 -0.01
      rotation 0 0 1 1.5708
      name "ds_left"
      lookupTable [
        0 0 0
        2 1000 0
      ]
      type "sonar"
      aperture 1.5709
    }
    DistanceSensor {
      name "ds_front"
      lookupTable [
        0 0 0
        2 1000 0
      ]
      type "sonar"
      aperture 1.5709
    }
    DistanceSensor {
      rotation 0 0 1 -1.5707953071795862
      name "ds_right"
      lookupTable [
        0 0 0
        2 1000 0
      ]
      type "sonar"
      aperture 1.5709
    }
    DistanceSensor {
      rotation 1.125079999996903e-14 -2.346409999993541e-06 0.9999999999972473 3.14159
      name "ds_back"
      lookupTable [
        0 0 0
        2 1000 0
      ]
      type "sonar"
      aperture 1.5709
    }
  ]
  cameraSlot [
    Camera {
      width 400
      height 240
      near 0.2
    }
  ]
}
